utils/utils.go

```diff
@@ -4,44 +4,69 @@ import (
 	"errors"
 	"fmt"
 	"io"
-	"path"
 	"regexp"
-	"runtime"
 	"strings"
 
-	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v5"
 )
 
 const (
-	// Windows constant represents a target operating system running a version of Microsoft Windows
-	Windows = "windows"
-	// BadFilePrefix constant is returned when path has leading slash or backslash
-	BadFilePrefix = "expecting only a filename prefix, which may not include slashes or backslashes"
+	// ErrBadAbsFilePath constant is returned when a file path is not absolute
+	ErrBadAbsFilePath = "absolute file path is invalid - must include leading slash and may not include trailing slash"
+	// ErrBadRelFilePath constant is returned when a file path is not relative
+	ErrBadRelFilePath = "relative file path is invalid - may not include leading or trailing slashes"
+	// ErrBadAbsLocationPath constant is returned when a file path is not absolute
+	ErrBadAbsLocationPath = "absolute location path is invalid - must include leading and trailing slashes"
+	// ErrBadRelLocationPath constant is returned when a file path is not relative
+	ErrBadRelLocationPath = "relative location path is invalid - may not include leading slash but must include trailing slash"
 )
 
-// regex to ensure prefix doesn't have leading '/', '.', '..', etc...
-var prefixCleanRegex = regexp.MustCompile("^[/.]*")
-
 // regex to test whether the last character is a '/'
 var hasTrailingSlash = regexp.MustCompile("/$")
 
-// AddTrailingSlash is a helper function accepts a path string and returns the path string with a trailing slash if
-// there wasn't one.
-func AddTrailingSlash(path string) string {
+// regex to test whether the first character is a '/'
+var hasLeadingSlash = regexp.MustCompile("^/")
 
-	runePath := []rune(path)
-	lastRune := runePath[len(runePath)-1]
+// RemoveTrailingSlash removes trailing slash, if any
+func RemoveTrailingSlash(path string) string {
+	return strings.TrimRight(path, "/")
+}
 
-	slash := "/"
-	if runtime.GOOS == Windows {
-		slash = "\\"
+// RemoveLeadingSlash removes leading slash, if any
+func RemoveLeadingSlash(path string) string {
+	return strings.TrimLeft(path, "/")
+}
+
+// ValidateAbsFilePath ensures that a file path has a leading slash but not a trailing slash
+func ValidateAbsFilePath(name string) error {
```

Comment:
Is there a reason "Rel" and "Abs" are abbreviated, when as a general rule all other parts of other function names are spelled out?